brave/brave-core
# Fixes some users are seeing a drop in estimated earnings after the upgrade

Resolves https://github.com/brave/brave-browser/issues/20868

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

See https://github.com/brave/brave-browser/issues/20868. However, we should ideally test upgrade paths with various 1.34.x data sets covering the following permutations:

`confirmations.json`/`transactions` node:
- No transactions
- Unreconciled transactions
- Reconciled transactions
- Unreconciled and reconciled transactions (that cross over this/last month)

`confirmations.json`/`payments` node:
- No payments
- Payment for this month
- Payment for last month
- Payment for this and last month

**NOTE:** Unreconciled transactions must be at the end of the list and have matching unblinded payment tokens in `confirmations.json`. `Current earnings this month (estimated)` should include unreconciled earnings for all months + reconciled earnings for this month. Tests should also rollover to the following month to confirm `Current earnings this month (estimated)` is updated to reflect unreconciled earnings for all months + reconciled earnings for this month.

Steps to reproduce:

- Fresh install
- Launch browser
- Quit browser
- Copy `ads_service` folder from 1.34.x to `Default` folder in the users profile

Make a backup of the user profile so that you can restore and test using the following steps (no need to run the above steps again)

- Launch browser
- Join Rewards
- Navigate to brave://rewards

EXPECTED RESULT: "Current earnings this month (estimated)", "Next payment date" and "Ads received this month" should match 1.34.x

- View an ad

EXPECTED RESULT: "Current earnings this month (estimated)" and "Ads received this month" should increase

- Restart the browser

EXPECTED RESULT: "Current earnings this month (estimated)" and "Ads received this month" should not decrease

- Redeem unblinded payment tokens

EXPECTED RESULT: `reconciled_at` column should have the reconciliation timestamp in the transactions database table for transactions which were successfully redeemed
